drivers/counter/counter_nrfx_rtc.c

```diff
@@ -72,18 +76,81 @@ static u32_t counter_nrfx_read(struct device *dev)
 	return nrfx_rtc_counter_get(&get_nrfx_config(dev)->rtc);
 }
 
+static int set_cc(struct device *dev, u8_t chan, u32_t val)
+{
+	const struct counter_nrfx_config *nrfx_config = get_nrfx_config(dev);
+	const nrfx_rtc_t *rtc = &nrfx_config->rtc;
+	NRF_RTC_Type  *reg = rtc->p_reg;
+	nrf_rtc_int_t int_mask;
+	nrf_rtc_event_t evt;
+	u32_t prev_val;
+	u32_t top;
+	u32_t now;
+
+	evt = (nrf_rtc_event_t)(offsetof(NRF_RTC_Type, EVENTS_COMPARE[chan]));
+	int_mask = (nrf_rtc_int_t)(NRF_RTC_INT_COMPARE0_MASK << chan);
+
+	nrf_rtc_event_clear(reg, evt);
+	top = counter_get_top_value(dev);
+	prev_val = nrf_rtc_cc_get(reg, chan);
+	nrf_rtc_cc_set(reg, chan, val);
+
+	/* If prev CC setting equals or is just in front of the counter then
+	 * there is a risk that before new CC will be set after enabling event
+	 * previous CC will generate COMPARE event. In such risk delay must be
+	 * introduced between writting CC value and enabling the event.
+	 */
+	now = nrf_rtc_counter_get(reg);
+	if (((prev_val - now) & top) == 1)
+	{
+		LOG_INST_INF(nrfx_config->log,
+			"Delay introduced due to risk of pre-firing.");
+		NRFX_DELAY_US(33);
+	}
+
+	nrf_rtc_event_enable(reg, int_mask);
+
+	/* If value is set to now + 1 wait one LFCLK cycle to ensure that it is
+	 * written to RTC register.
+	 */
+	if (((val - now) & top) == 1) {
+		LOG_INST_INF(nrfx_config->log,
+			"Delay introduced due to short CC.");
+		NRFX_DELAY_US(33);
```

Comment:
33 used in more than one place => define.

Reply:
fixed